### PR TITLE
TST: new set of tests 

### PR DESCRIPTION
**Pull request by lucasfourier**
## Pull request type

- [ ] Code changes (bugfix, features)
- [x] Code maintenance (refactoring, formatting, tests)
- [ ] ReadMe, Docs and GitHub updates
- [ ] Other (please describe):

## Checklist

- [x] Tests for the changes have been added (if needed)
- [x] Docs have been reviewed and added / updated
- [x] Lint (`black rocketpy/ tests/`) has passed locally 
- [ ] All tests (`pytest --runslow`) have passed locally

## Current behavior
Tests won't pass with --runslow locally due to the following error:
"_tkinter.TclError: Can't find a usable tk.tcl in the following directories:"
which occurs non-deterministically. However, the newly implemented
tests passed locally with --runslow.

Enter text here...

## New behavior
The test files test_flight, test_rocket and test_function have gained
new unit tests. 

## Breaking change

- [ ] Yes
- [x] No

## Additional information

Enter text here...

### CODE REVIEW

File: tests/test_rocket.py [hunk 1]
```diff
@@ -489,6 +490,28 @@ def test_add_cp_eccentricity_assert_properties_set(calisto):
...
+    calisto_motorless : Rocket instance
+        A predefined instance of a Rocket without a motor, used as a base for testing.
+    cesaroni_m1670 : rocketpy.SolidMotor
+        Cesaroni M1670 motor
+    """
+    assert isinstance(calisto_motorless.motor, EmptyMotor)
+    center_of_mass_motorless = calisto_motorless.center_of_mass
+    calisto_motorless.add_motor(cesaroni_m1670, 0)
+
+    assert calisto_motorless.motor is not None
```
Comment (Gui-FernandesBR) [inline tests/test_rocket.py:507]: More import to check if it is an instance of the motor class.

Please remeber to import the Motor class at the top of the file.

```suggestion
    assert isinstance(calisto_motorless.motor, Motor)
```

File: tests/test_rocket.py [hunk 2]
```diff
@@ -509,3 +532,25 @@ def test_set_rail_button(calisto):
...
+
+def test_evaluate_total_mass(calisto_motorless):
+    """Tests the evaluate_total_mass method of the Rocket class.
+    Both with respect to return instances and expected behaviour.
+    Parameters
+    ----------
+    calisto_motorless : Rocket instance
+        A predefined instance of a Rocket without a motor, used as a base for testing.
+    """
+    assert isinstance(calisto_motorless.evaluate_total_mass(), Function)
```
Comment (Gui-FernandesBR) [inline tests/test_rocket.py:545]: Good idea to assert this part of the code. But I'd suggest to also check for expected values.

**Review by Gui-FernandesBR — APPROVED** [review]
Nice work here, and very nice to see contributions from you again @lucasfourier !!

I made some tiny comments but overall the code is already improving our testing suit. Well done!

File: tests/test_function.py [hunk 3]
```diff
@@ -97,12 +97,17 @@ def test_interpolation_methods(linear_func):
...
     assert linear_func.get_interpolation_method() == "akima"
     assert np.isclose(linear_func.get_value(0), 0.0, atol=1e-6)
 
     # Test polynomial
+
+    assert isinstance(linear_func.set_interpolation("polynomial"), Function)
     linear_func.set_interpolation("polynomial")
+    assert isinstance(linear_func.get_interpolation_method(), str)
     assert linear_func.get_interpolation_method() == "polynomial"
     assert np.isclose(linear_func.get_value(0), 0.0, atol=1e-6)
```
[the remark this answers is not in the document]
Comment (lucasfourier) [reply tests/test_function.py:112]: Yes! In the near future this will be done.

File: tests/test_rocket.py (hunk 1 above)
Comment (lucasfourier) [reply tests/test_rocket.py:507]: Also agree, done it.

**Pull request merged**